Update workflow retry docs with some clarifications (#4936)
dapr/docs: daprdocs/content/en/developing-applications/building-blocks/workflow/workflow-features-concepts.md

| Parameter | Description |
| --- | --- |
| **Maximum number of attempts** | The maximum number of times to execute the activity or child workflow. If set to 0, no attempts will be made. |
| **First retry interval** | The amount of time to wait before the first retry. |
| **Backoff coefficient** | The coefficient used to determine the rate of increase of back-off. For example a coefficient of 2 doubles the wait of each subsequent retry. |
| **Maximum retry interval** | The maximum amount of time to wait before each subsequent retry. If set to 0, no retries will happen. |

> | **Retry timeout** | The global timeout for retries, regardless of any configured max number of attempts and retries. |

Reviewer: ```suggestion
| **Retry timeout** | The global timeout for retries, regardless of any configured max number of attempts.  No further attempts will be made at executing a activities after this timeout expires.
```

> | **Retry timeout** | The global timeout for retries, regardless of any configured max number of attempts.  No further attempts will be made at executing a activities after this timeout expires.

Reviewer 2: ```suggestion
| **Retry timeout** | The global timeout for retries, regardless of any configured max number of attempts.  No further attempts are made executing activities after this timeout expires.
```
